### PR TITLE
Use tool.scikit-build.cmake.version, set scikit-build-core minimum-version

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to https://github.com/rapidsai/build-planning/issues/58.

`scikit-build-core==0.10.0` was released today (https://github.com/scikit-build/scikit-build-core/releases/tag/v0.10.0), and wheel-building configurations across RAPIDS are incompatible with it.

This proposes upgrading to that version and fixing configuration here in a way that:

* is compatible with that new `scikit-build-core` version
* takes advantage of the forward-compatibility mechanism (`minimum-version`) that `scikit-build-core` provides, to reduce the risk of needing to do this again in the future

### CODE REVIEW

**Comment by jameslamb** [comment]
The `devcontainers` build should be fixed once https://github.com/rapidsai/devcontainers/pull/377 is merged. Otherwise, this is ready for review.

File: python/cudf/pyproject.toml [hunk 1]
```diff
@@ -133,7 +133,8 @@ requires = [
 [tool.scikit-build]
 build-dir = "build/{wheel_tag}"
 cmake.build-type = "Release"
-cmake.minimum-version = "3.26.4"
+cmake.version = "CMakeLists.txt"
+minimum-version = "0.10"
```
Comment (jameslamb) [inline python/cudf/pyproject.toml:137]:
```suggestion
minimum-version = "build-system.requires"
```

**Comment by jakirkham** [comment]
Restarting failed CI jobs now that `devcontainers` have been updated

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks James! 🙏

**Pull request merged**